### PR TITLE
Adds initial openshift/boilerplate onboarding configs

### PR DESCRIPTION
**Pull request by clcollins**
This commit:

* Adds the Boilerplate update mechanism to allow for bootstrapping the rest of the project.
* Initializes the boilerplate setup with `make boilerplate-update` following the quickstart onboarding documentaton.
* Adds codecov.io config and badge

NOTE:  This depends on #506, and must merge after that.

REF:
[https://issues.redhat.com/browse/OSD-5822](https://issues.redhat.com/browse/OSD-5822)

Signed-off-by: Christopher Collins <collins.christopher@gmail.com>

### CODE REVIEW

**Review by 2uasimojo — COMMENTED** [review]
This is a great start. A couple of the issues noted inline will choke CI until fixed; the rest are pretty much cleanup. Thanks!

File: build/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM registry.svc.ci.openshift.org/openshift/release:golang-1.13 AS builder
+FROM quay.io/app-sre/boilerplate:image-v0.3.0 AS builder
 
 ENV OPERATOR_PATH=/go/src/github.com/openshift/aws-account-operator
 ENV GO111MODULE=on
```
[the remark this answers is not in the document]
Comment (2uasimojo) [reply build/Dockerfile:4]: (This is both of your CI fails btw)

File: boilerplate/openshift/golang-osd-operator/ensure.sh [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        fi
+    fi
+    # Create (or overwrite) the symlink to the binary we discovered or
+    # downloaded above.
+    ln -sf $osdk operator-sdk
+    ;;
+
+venv)
+    # Set up a python virtual environment
+    python3 -m venv .venv
```
[the remark this answers is not in the document]
Comment (clcollins) [reply boilerplate/openshift/golang-osd-operator/ensure.sh:86]: Done!

File: boilerplate/openshift/golang-osd-operator/ensure.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    # Ensure operator-sdk is installed at the desired version
+    # When done, ./.operator-sdk/bin/operator-sdk will be a
+    # symlink to the appropriate executable.
+    #########################################################
+    # First discover the desired version from go.mod
+    # The following properly takes `replace` directives into account.
+    wantver=$(go list -json -m github.com/operator-framework/operator-sdk | jq -r 'if .Replace != null then .Replace.Version else .Version end')
+    echo "go.mod says you want operator-sdk $wantver"
+    # Where we'll put our (binary and) symlink
+    mkdir -p .operator-sdk/bin
```
[the remark this answers is not in the document]
Comment (clcollins) [reply boilerplate/openshift/golang-osd-operator/ensure.sh:42]: Done!

File: Makefile [hunk 4]
```diff
@@ -8,9 +8,12 @@ include hack/scripts/test_envs
...
-include project.mk
-include standard.mk
+# Boilerplate
+include boilerplate/generated-includes.mk
+
+.PHONY: boilerplate-update
+boilerplate-update:
+	@boilerplate/update
 
 default: gobuild
```
[the remark this answers is not in the document]
Comment (clcollins) [reply Makefile]: Done, done and done!

File: Makefile [hunk 5]
```diff
@@ -8,9 +8,12 @@ include hack/scripts/test_envs
...
```
[the remark this answers is not in the document]
Comment (clcollins) [reply Makefile]: Done!

File: build/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (clcollins) [reply build/Dockerfile:4]: Done!  And looks like the CI jobs are working now.

**Comment by 2uasimojo** [comment]
Javol, this should do the trick. Thanks for the updates @clcollins!

/lgtm

**Pull request merged**